walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,54 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Get the attributes of the requested meeting.
+    Collection<String> requestedAttendees = request.getAttendees();
+    int requestedDuration = (int) request.getDuration();
+
+    // Get all the conflicting times based on the requested attendees and their planned events.
+    Collection<TimeRange> busyTimes = new HashSet<>();
+    for (Event event : events) {
+      Set<String> attendees = event.getAttendees();
+      for (String requestedAttendee : requestedAttendees) {
+        if (attendees.contains(requestedAttendee)) {
+          busyTimes.add(event.getWhen());
+          break;
+        }
+      }
+    }
+
+    // Get all the times of the requested duration or longer that have no overlap with a conflicting time.
+    Collection<TimeRange> freeTimes = new ArrayList<>();
+    int startTime = TimeRange.START_OF_DAY;
+    int duration = requestedDuration;
+    while (startTime + duration <= TimeRange.END_OF_DAY + 1) {
+      TimeRange testTime = TimeRange.fromStartDuration(startTime, duration);
+      boolean hasConflict = false;
+      for (TimeRange busyTime : busyTimes) {
+        if (testTime.overlaps(busyTime)) {
+          hasConflict = true;
+          duration = busyTime.start() - startTime;
```

Comment:
use a different variable here instead of temporarily assigning duration and then reassigning it in L55